lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2842,7 +2842,14 @@ static ArrayRef<Decl *> evaluateMembersRequest(
     if (auto *vd = dyn_cast<ValueDecl>(member)) {
       // Add synthesized members to a side table and sort them by their mangled
       // name, since they could have been added to the class in any order.
-      if (vd->isSynthesized()) {
+      if (vd->isSynthesized() &&
```

Comment:
This is a suggestion for main and not 5.9: isSynthesized() was actually originally added for this exact case, so it might be better to set the flag when synthesizing those properties in the first place.